### ISSUE TITLE
Message type "caffe.LayerParameter" has no field named "region_param".

### ISSUE THREAD

**Issue by dymat** [ysh329/darknet-to-caffe-model-convertor#1]
Hi there!

Thanks for your conversion tool! I used it for converting the darknet tiny-yolo-voc model (v2) into caffe. It worked without any problems.

However, at inference I get this error: Message type "caffe.LayerParameter" has no field named "region_param". Here's my generated prototxt: https://pastebin.com/JBiyQjQg

How can I use tiny yolo v2 as caffemodel? Do you use a special caffe version?

Thanks in advance,
~dymat

**Comment by ysh329** [comment]
Well, you're right. A special caffe version, which supports `Region` layer, is necessary. In fact, I only use caffe model as an intermediate model.

**Comment by bharathbv** [comment]
@ysh329  appreciate all your contribution. I am encountering similar problem. After converting yolo_v2 to caffemodel, I am not able to use them on caffe. I am seeing similar issue  
 Message type "caffe.LayerParameter" has no field named "region_param"

Are you aware of special Caffe version?

**Comment by ysh329** [comment]
Eh, I only make caffe model as intermediate model. However, you can check this repo : https://github.com/gklz1982/caffe-yolov2

**Comment by youngxiao** [comment]
@dymat The same issue with you. Have you solve it or just use caffemodel as anintermediate model?

**Comment by dymat** [comment]
hi @youngxiao,

I solved it by removing the complete region layer. Then I reimplemented the logic originally performed by the region layer. I.e. I take the logits from the layer before the region layer and interpret these as bounding boxes. 

Here's my implementation in GO: https://github.com/dymat/GOLOv2
The region logic is here: https://github.com/dymat/GOLOv2/blob/master/yolov2RegionLayer.go

Hope it helps.
D

**Comment by youngxiao** [comment]
Thanks, @dymat I will try your implementation.
